src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -372,7 +373,7 @@ void ov::npuw::JustInferRequest::bind_global_parameters(std::size_t idx) {
     auto& comp_model_desc = m_npuw_model->m_compiled_submodels[idx];
     const auto real_idx = comp_model_desc.replaced_by.value_or(idx);
 
-    const bool do_copy = needs_copy(idx);
+    const bool do_copy = !m_alloc_required && needs_copy(idx);
```

Comment:
Not sure if it's correct

Reply:
Here we check global parameters. The idea is that global parameters allocation solely depends on `m_alloc_required` - if it's set, params will be allocated on NPU

Reply:
I think `m_alloc_required` is overall misleading. It is always required.

Also, you've missed this in my past L0 PR: https://github.com/dmatveev/openvino/pull/5

More precisely, this part: https://github.com/dmatveev/openvino/blob/e7d62f1a4412f639d0fb112e4f5647eeff9a1b8e/src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp#L117

And then this part: https://github.com/dmatveev/openvino/blob/e7d62f1a4412f639d0fb112e4f5647eeff9a1b8e/src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp#L370

The backstory here is that, even if you've allocated your model-global input tensors yourself, they _may_ be overwritten. Even our scripts carelessly do this, unfortunately. So what you need to do is to keep track of the tensors you allocate (maybe you can just memorize the pointers in your new `allocTensor` method) and check if the tensors' your working with are still "known" to you.

Once there's a `set_tensor` call - you loose it and your `m_alloc_required` flag doesn't tell truth.

Reply:
Good point, thanks